### ISSUE TITLE
Add option to filter out links

### ISSUE THREAD

**Issue by Robbepop** [Aaronepower/tokei#227]
Today I found out that tokei's reported lines of code for my repository seemed to explode.
The reason for this was that tokei (incorrectly) counted links (symbolic-links and hard-links) as if they were regular files.

I think it would be the best to make tokei completely ignore symbolic links and ignore all multiple occurrences of the same hard linked file.

While restating defaults might be hard, it should be also okay to add options to filter links out of the accumulation process.
I think options like
- `--ignore-symbolic-links`: Ignores any symbolic link.
- `--ignore-multiple-hard-links`: Ignores all but one occurences of a hard linked file that has its source within the given search space (repository for example) and ignore all hard linked files that have their source outside of the search space.

**Comment by jhpratt** [comment]
@XAMPPRocky How difficult would this be to implement? Without having looked at tokei (aside from adding PostCSS), my initial thought would be to use a `HashSet` to store inodes as files are visited. I'd have to check on OS-interop for that, though.

**Comment by XAMPPRocky** [comment]
@jhpratt Hey, I don't about about the second option, however the first option should be easy to add.  Tokei's CLI is set using `clap` in [`src/cli.rs`](https://github.com/XAMPPRocky/tokei/blob/master/src/cli.rs) and its configuration in [`src/config.rs`](https://github.com/XAMPPRocky/tokei/blob/2129e27d6a69320b19b3e51fac590dd5a74ffde5/src/config.rs) I would create this option very similar to `--no-ignore`.

For traversing the file system tokei uses `ignore` which has an option in `WalkBuilder::follow_links` that says whether it should follow symbolic links or not.

**Comment by jhpratt** [comment]
Looks like there's the [`same-file`](https://github.com/BurntSushi/same-file) crate which could be of use. A quick test shows it handles both hard and symbolic links correctly.

Looking through tokei's code, I presume `src/utils/fs.rs` is where a change would need to be made, given the presence of the `get_all_files` method. I'm not entirely sure what is happening there, though.

**Comment by XAMPPRocky** [comment]
@jhpratt Yes sorry I should have pointed you to that. I believe that is what `ignore` uses for `follow_links`. I would add the `follow_links` as part of the [`WalkerBuilder` configuration](https://github.com/XAMPPRocky/tokei/blob/master/src/utils/fs.rs#L51-L57)

Everything before `walker.build_parallel().run(…)` is configuring the walker's behaviour (what paths to search, to ignore, etc). `walker.build_parallel().run(…)` runs in parallel over the paths and sends any file paths to `rx` (a `crossbeam::Receiver`), which are then [validated as programming languages](https://github.com/XAMPPRocky/tokei/blob/master/src/utils/fs.rs#L93-L108).

**Comment by jhpratt** [comment]
Looking through various documentation, looks like symlinks are currently ignored, and a trivial check shows this is the case.

What would be the preferred way to handle hard links? If it were up to me, I'd lean towards automatically excluding anything past the first, as you'd essentially be counting the file twice.
